st3/sublime_lib/view_stream.py

```diff
@@ -1,21 +1,32 @@
 from sublime import Region
-from io import SEEK_SET, SEEK_CUR, SEEK_END
+from io import SEEK_SET, SEEK_CUR, SEEK_END, TextIOBase
 
 
-class ViewStream():
+class ViewStream(TextIOBase):
+    """A writable text stream encapsulating a `sublime.View` object."""
+
     def __init__(self, view):
         self.view = view
 
     def _check_selection(self):
         if len(self.view.sel()) == 0:
             raise ValueError("The underlying view has no selection.")
-        if len(self.view.sel()) > 1:
+        elif len(self.view.sel()) > 1:
             raise ValueError("The underlying view has multiple selections.")
+        elif not self.view.sel()[0].empty():
+            raise ValueError("The underlying view's selection is not empty.")
 
     def _check_is_valid(self):
         if not self.view.is_valid():
             raise ValueError("The underlying view is invalid.")
 
+    """Insert the string <var>s</var> into the view and return the number of
```

Comment:
docstrings should to go inside the method definition.

I've never seen `<var>s</var>` used before (in Python at least). Does this follow a standard and, if so, which?

Reply:
That's HTML. It might not make any sense in a pydoc. Is there a good reference somewhere for pydoc conventions?